src/index.js

```diff
@@ -5,7 +5,23 @@ const Input = require("./input.js")
 const { getPackages, deletePackage } = require("./query.js")
 const { process } = require("./process.js")
 
+const DEFAULT_VERSION_PATTERN = /^.+$/
+const DEFAULT_KEEP = 2
+
 function getAllInput() {
+  // Use `version` to determine two kinds of logic throughout.
+  const version = getInput("version") ? getInput("version") : null
+
+  // Ignore keep and version-pattern settings if version is set
+  let keep, versionPattern
+  if (version) {
+    keep = 0
+    versionPattern = DEFAULT_VERSION_PATTERN
+  } else {
+    keep = getInput("keep") ? getInput("keep") : DEFAULT_KEEP
+    versionPattern = getInput("version-pattern") ? getInput("version-pattern") : DEFAULT_VERSION_PATTERN
+  }
```

Comment:
Let's make the action crash with an error when incompatible settings are used together. This way we make sure that the action is not misused unintentionally and hopefully reduce confusion.